tox.ini

```diff
@@ -141,19 +141,10 @@ commands =
 [testenv:pep8]
 basepython = python3
 deps=
-      flake8
-      flake8-import-order
-      flake8-builtins
-      flake8-docstrings>=1.3.1
-      flake8-rst-docstrings
-      # flake8-rst-docstrings dependency, leaving it here
-      # in case it requires a version pin
-      pydocstyle
-      pygments
-      black==21.5b1
+      pre-commit
 commands =
-     flake8 ./lib/ ./test/ ./examples/ setup.py doc/build/conf.py {posargs}
-     black --check ./lib/ ./test/ ./examples/ setup.py doc/build/conf.py
+     pre-commit run --all-files --show-diff-on-failure flake8
```

Comment:
this doesn't modify the .git checkout at all, right?

Reply:
The Black one can, I'll add back `--check` to match the earlier behaviour.

Reply:
Ah wait we can't do that, the Black specific argument would need to go in `.pre-commit-config.yaml` and so affect the normal pre-commit use.

Would you prefer I revert the last commit and update the pins in `tox.ini`?

Reply:
I think so unless pre-commit has a way to add arguments to individual hooks when you run them on the commandline.

tox has to just be able to run without mutating anything, obviously.

Reply:
I personally don't mind using tox to also run formatters :)

Anyway, I've updated the pins in tox.ini instead to keep it consistent.

Reply:
running tox to run formatters is fine but that would be a different tox task here.   thanks for updating!